platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/State.java

```diff
@@ -158,6 +159,12 @@ public PlatformState getWritablePlatformState() {
         return getChild(ChildIndices.PLATFORM_STATE);
     }
 
+    @Override
+    public WritableRosterStore getWritableRosterStore() {
+        // roster is only a part of the platform state
+        throw new UnsupportedOperationException("Roster can only be obtained from the state used by the platform");
```

Comment:
Why should this be the case? The concept of platform state isn't really super relevant anymore with the upcoming consensus/execution code split.

Reply:
I don't think this implementation is required by apps that use this class as they will get their rosters by trivially calling `createRoster(AddressBook)`.

Reply:
I'm not following this reasoning.

It seems that the writeable roster state would fall into the same category as the writeable platform state, which *is* accessible from this class. The discrepancy between these 2 states is what I'm not understanding

Reply:
As I understand today:
1. There are two types of state data provisioned in this class - swirld state and "platform side" state. `Rosters` are a kind of platform side state that just happens to have its own store. 
2. As far as I can tell, perhaps most importantly, none of the test Mains that implement this deprecated class needs to modify a Roster from the state (their roster is obtained by calling `createRoster` as can be seen in the `Browser` class) so no need to provide a roster store here.

Reply:
@litt3, this class is deprecated and no longer used in production code. Granted, the structure is very confusing today because the work to update the testing and demo apps was delayed in order to prioritize getting the State API moved into the platform. The real class to use is `MerkleStateRoot`, which implements both the `State` interface (the new State API we are moving toward) and the old `MerkleRoot` interface (which we want to eliminate). The cleanup work will be done soon.